### DISCUSSION TITLE
Manipulation of form fields in Extracted Form Object "must not be accessed before initialization"

### DISCUSSION THREAD

**Discussion by idealerror** [livewire/livewire#7497] [1. Help] [answered]
Hello,

I'm trying to understand how to use Livewire Form classes.  I need to be able to manipulate fields of the form as users select or fill in different options of the form.  

In the Livewire Component, I'm trying to adjust a form field and I'm getting the error that you cannot adjust a field before it's been set.  In this case, the user inputs their `form.email` first and I'm trying to extract part of the email and put it into the `form.name` field.  When doing that, I get an error that I cannot access `FormName::$name`.

```
FormName::$name must not be accessed before initialization
```

How can I manipulate form fields when extracting all the wire:models to the Form class?

**Comment by joshhanley** [comment]
@idealerror please share your code 🙂

**Reply by idealerror** [reply]
Sharing the relevant parts below as there's a ton of unnecessary stuff. 

When a user inputs an email address into the form (`wire:model.live="form.email"`), it should update the model "`form.name`" with a value.  Instead, I'm getting the error that `$name` is not initialized yet. 

*Component*
```php
class TestComponent extends Component 
{
    public FormName $form;

    public function updatedFormEmail($value)
    {
        $this->form->name = 'test';
    }
}
```

**Form**
```php
class FormName extends Form 
{
    public string $email;
    public string $name;

    public function mount(){
        $this->name = '';
    }
}
```

**Reply by joshhanley** [reply, marked as answer]
@idealerror so the issue is the typed properties `$email` and `$name` are not initialised.

So just do this:

```php
public string $email = '';
public string $name = '';
```

And that should fix the issue.

There is no mount (or any lifecycle hooks) for Form Objects. So if you don't want to initialise as I mentioned above, you will need to do it in mount of your component, or call `$this->form->mount();` from your component mount.

Hope this helps!

**Reply by ph7jack** [reply]
I think I have a related issue https://github.com/livewire/livewire/discussions/8473. 
What do you think @joshhanley?
